pkg/engine/context/evaluate.go

```diff
@@ -54,10 +54,15 @@ func (ctx *context) getMatchingLoaders(query string) []DeferredLoader {
 	defer ctx.deferred.mutex.Unlock()
 
 	var matchingLoaders []DeferredLoader
-	for name, deferredLoader := range ctx.deferred.loaders {
+	for _, name := range ctx.deferredOrder {
 		if strings.Contains(query, name) {
-			matchingLoaders = append(matchingLoaders, deferredLoader)
-			delete(ctx.deferred.loaders, name)
+			matchingLoaders = append(matchingLoaders, func() error {
+				if deferredLoader, ok := ctx.deferred.loaders[name]; ok {
+					delete(ctx.deferred.loaders, name)
+					return deferredLoader()
+				}
+				return nil
+			})
```

Comment:
TBH honest i would prefer this func to return just the matching loader names, and an evaluate loader func to do the actual deletion/processing.

Embedding the deletion in the loader was just a hack :)

Reply:
Okay, let me give that a try instead